robot/impl/local_robot.go

```diff
@@ -1174,13 +1174,6 @@ func (r *localRobot) applyLocalModuleVersions(cfg *config.Config) {
 }
 
 func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config, forceSync bool) {
-	r.configRevisionMu.Lock()
```

Comment:
Is the consequence here that we now only report a newer config revision after* package downloading succeeds?

Also I don't see the configRevisionMu being locked. Is that important?

Reply:
the missing lock is a miss, added it back in.

thinking about it more, the config revision should update at the top of the function so users will know the new config did get pulled and is being processed. In theory, it may make sense to rollback the revision if we exit reconfiguration if download fails, but I also think it's ok to keep the new revision as a sign that the new config did get loaded. whether we revert the revision or not, it will be unclear what state robot reconfiguration is in, so maybe we need to add more information in there

Reply:
I know @maximpertsov  and I talked about how to represent "mixed" state when a new revision is being applied/doesn't apply completely. 

> so maybe we need to add more information in there

I agree the only path to do better is to add more information. And I agree it's best to do that thinking/work in a separate ticket.